### ISSUE TITLE
[DEMO] Hybrid quantum-classical auto encoder

### ISSUE THREAD

**Issue by sophchoe** [PennyLaneAI/qml#341]
#### General information

**Name**
Sophie Choe

**Affiliation** (optional)
Portland State University
--------------------------------------------------------------------------------

#### Demo information

**Title**
Hybrid quantum-classical auto encoder

**Abstract**
Keras auto-encoder hybrid model 
Encoder: 6 Keras dense layers
Decoder: 3 Pennylane qml layers 

The loss function in the paper requires state vectors and probability, however Pennylane measurement module does 
not support state vector retrieval (as of 9/24/2021). Hence mean squared error from Keras is used instead on the 
output vectors of probability measurement. To get vectors of length 3, the cutoff dimension parameter is set to 3. 

**Relevant links**
https://github.com/sophchoe/QML/blob/main/auto_encoder_Pennylane_Keras.ipynb
https://arxiv.org/pdf/1806.06871v1.pdf

**Comment by CatalinaAlbornoz** [comment]
Thank you @sophchoe for submitting this Demo!

**Comment by sophchoe** [comment]
Hi Catalina,

I cleaned up the code slightly and reposted.

The paper called for 25 quantum layers which I thought was only 3.

Upon running an experiment with 25 layers, I saw a very slight improvement.

Thanks.

**Comment by CatalinaAlbornoz** [comment]
Hi @sophchoe, thank you for this improvement! I hope that we can get the demos published today. I will let you know when they are up!

**Comment by CatalinaAlbornoz** [comment]
Hi @sophchoe. The Demo is up! You can see it here https://pennylane.ai/qml/demos_community.html
Thank you once again for this contribution!
